R/utils-data.R

```diff
@@ -217,6 +217,16 @@ get_data <- function(all_data, name) {
 return_data <- function(...) {
   dots <- list(...)
   names(dots) <- as.list(substitute(list(...)))[-1L]
+  # disallow any data which is "grouped" as it may lead to unexpected
+  # behavior, especially for unsuspecting chunks which may use it down
+  # the line not expecting any groupings.
+  lapply(names(dots), function(dname) {
+    # note we wrap the assert in a tryCatch since groups will throw an
```

Comment:
Yea, I think so.  When I was looking at the documentation for `group_by` it sounded like it supported many different types of tables and when I check `is_tibble(data.table::data.table())` it returned false.  Thus I wasn't sure how to check all possible types and so put in the `tryCatch`. 
But as I try `is_tibble(data.table::data.table(...) %>% group_by(x))` it sure enough does report `true`.  And of course if it didn't get grouped then we don't really need to do the check anyway.
I'll switch that and update the comments.